src/widget/wfindonwebmenu.cpp

```diff
@@ -0,0 +1,269 @@
+        TrackSearchProperties trackSearchProperty) {
+    const QString prefixAction = composePrefixAction(trackSearchProperty);
+    switch (service) {
+    case Service::SoundCloud: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, albumName),
+                this,
+                [this, service, trackSearchProperty, albumName]() {
+                    emit triggerBrowser(service, trackSearchProperty, albumName);
+                });
+        break;
+    }
+
+    case Service::LastFm: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, albumName),
+                this,
+                [this, service, trackSearchProperty, albumName]() {
+                    emit triggerBrowser(service, trackSearchProperty, albumName);
+                });
+        break;
+    }
+
+    case Service::Discogs: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, albumName),
+                this,
+                [this, service, trackSearchProperty, albumName]() {
+                    emit triggerBrowser(service, trackSearchProperty, albumName);
+                });
+        break;
+    }
+    default:
+        break;
+    }
+}
+
+void WFindOnWebMenu::addActionsTrackTitle(Service service,
+        const QString& trackTitle,
+        QMenu* pServiceMenu,
+        TrackSearchProperties trackSearchProperty) {
+    const QString prefixAction = composePrefixAction(trackSearchProperty);
+    switch (service) {
+    case Service::SoundCloud: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, trackTitle),
+                this,
+                [this, service, trackSearchProperty, trackTitle]() {
+                    emit triggerBrowser(service, trackSearchProperty, trackTitle);
+                });
+        break;
+    }
+
+    case Service::LastFm: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, trackTitle),
+                this,
+                [this, service, trackSearchProperty, trackTitle]() {
+                    emit triggerBrowser(service, trackSearchProperty, trackTitle);
+                });
+        break;
+    }
+
+    case Service::Discogs: {
+        pServiceMenu->addAction(
+                composeActionText(prefixAction, trackTitle),
+                this,
+                [this, service, trackSearchProperty, trackTitle]() {
+                    emit triggerBrowser(service, trackSearchProperty, trackTitle);
+                });
+        break;
+    }
+    default:
+        break;
+    }
+}
+
+void WFindOnWebMenu::openInBrowser(Service service,
+        TrackSearchProperties trackSearchProperty,
+        const QString& query) {
+    const QString serviceUrl = getServiceUrl(service, trackSearchProperty);
+    QUrlQuery urlQuery;
+    urlQuery.addQueryItem("q", query);
+    QUrl url(serviceUrl);
+    url.setQuery(urlQuery);
+    QDesktopServices::openUrl(url);
```

Comment:
since the failure is unexpected, using Verify_or_Drbugassert() should be used.

Reply:
However. But at least a warning log message should be logged to allow tracing unexpected failures.

Reply:
What would be the best use-case in this scenario? is qWarning okay, or should qDebug be used? Should URL be in the log warning?

Reply:
Oh I see the debug assertion is a already above. I did some test and I was not able to fail openUrl() even with obscure urls. 

So I guess an error only happens when the users system is miss-configured.
https://github.com/qt/qtbase/blob/42e13b7c61693ca95e119106d4f6dbd2bcf2308d/src/gui/util/qdesktopservices.cpp#L195 
And many of those cases have already a qWarning when digging deeper. 

Conclusion: A normal qWarning() including the URL is sufficient. 
Maybe just technical: 
"QDesktopServices::openUrl() failed for <URL>"

Reply:
I have tried to check URL's in two different ways.
First with [QUrl::isValid()](https://doc.qt.io/qt-6/qurl.html#isValid) which returns always false because the propoerties have white spaces and they are not encoded, that is because it is always returning false.
Second is (Which I used for now) [QUrl::isEmpty()](https://doc.qt.io/qt-6/qurl.html#isEmpty) which returns always true because the URL is not empty at all. But IDK if this one makes wanted validation.

My question is should I keep using isEmpty() with existing configuration, can it make validation or should I make encoding with white spaces and use isValid()? 

---

While I was writing my question, I have tried it again, I realised that isValid() is returning true with the existing URL's which everything was fine in the first place. Then I realised that VERIFY_OR_DEBUG_ASSERT() is executed if the condition is false, so I should of use it like VODA(isValid()) instead of VODA(!isValid())

So long story short, I guess, It's better to use isValid() with the technical qWarning().

Reply:
All URLs produced by this code must be valid. Anything else would be a programming error, hence a DEBUG_ASSERT without a log is appropriate.

But QDesktopServices::openUrl() might fail for other reasons out of our control when launching the application, even if the URL is valid.

Reply:
I think right now with the last commits URL is validated. 

And for the QDesktopServices::openUrl() we can not do anything, right?

Reply:
Please log a warning message inlcuding the URL as already suggested, wrapped into VERIFY_AND_DEBUG_ASSERT.